metals/src/main/scala/scala/meta/internal/ansi/LineListener.scala

```diff
@@ -46,6 +46,8 @@ class LineListener(onLine: String => Unit) {
     )
   }
 
+  def appendLine(text: String): this.type = appendString(text + "\n")
```

Comment:
If you look at the scaladoc of this class you'll see

>  * Converts a stream of strings (with potential ANSI codes and newlines) into a callback for indvidual lines where each individual line has no newline \n characters.

It's sort of confuses me now a bit since we say that there is no newline characters, but then we have a method that adds a newline character.

Reply:
Honestly,  I wanted to avoid touching it :smile: But, yea, you right - it's confusing.

I reduced `LineListener` to `AnsiFilter` and dropped existing logic around handling newlines. It was used only for process handling and now it's readers split output by LF automatically.

`AnsiFilter` might look a little bit weird because of having an internal variable but I took into account [these notes](https://github.com/scalameta/metals/blob/16806aa5674c0656b7c4e634dcf882dbddffe00a/metals/src/main/scala/scala/meta/internal/ansi/LineListener.scala#L50-L59) about it's performance and tried to leave it comparable to the previous implementation.

Reply:
> AnsiFilter might look a little bit weird because of having an internal variable but I took into account these notes about it's performance and tried to leave it close to the previous implementation.

👍🏼 makes sense.